packages/vite/src/node/http.ts

```diff
@@ -129,16 +129,16 @@ export async function resolveHttpsConfig(
   const httpsOption = isObject(https) ? https : {}
```

Comment:
Maybe we could clone it upfront and then we can leave the rest of the code untouched?
```suggestion
  const httpsOption = isObject(https) ? { ...https } : {}
```